### PR TITLE
ELPP-3588 Update PHP image and use Docker Compose

### PR DESCRIPTION
**Pull request by thewilkybarkid**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,12 @@
-FROM elifesciences/php_cli
+ARG image_tag=latest
+FROM elifesciences/proofreader-php_composer:${image_tag} AS build
+FROM php:7.0.29-cli-alpine
```
Comment (thewilkybarkid) [inline Dockerfile]: Deliberately not an eLife image, to reduce the size.

File: Dockerfile [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
 
-USER root
-RUN touch .php_cs.cache && chown www-data:www-data .php_cs.cache
+ENV PROJECT_FOLDER=/srv/proofreader-php
+RUN mkdir ${PROJECT_FOLDER}
+WORKDIR ${PROJECT_FOLDER}
 
-USER www-data
-CMD bin/proofreader
+COPY bin/ bin/
```
Comment (giorgiosironi) [inline Dockerfile]: I think this will be owned by `root`, which would still make it read only for the `www-data` user

Comment (giorgiosironi) [reply Dockerfile]: In general the use of builder images should make the `elife` user less important as they have their own user for running processes like `composer`

File: Dockerfile.ci [hunk 3]
```diff
@@ -0,0 +1,22 @@
+ARG image_tag=develop
+FROM elifesciences/proofreader-php_composer_dev:${image_tag} AS build
+FROM elifesciences/php_7.0_cli:656bb4bdf1e49a5e80337e2a7c4f44f10c3f52b0
```
Comment (giorgiosironi) [inline Dockerfile.ci]: the base image version should probably start to be moved to `.env`? ease of update

File: Dockerfile.ci [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+USER root
+
+WORKDIR ${PROJECT_FOLDER}
+RUN touch .php_cs.cache && \
+    chown elife:elife . && \
+    chown www-data:www-data .php_cs.cache
+
+USER elife
+
+COPY --chown=elife:elife .php_cs ./
```
Comment (giorgiosironi) [inline Dockerfile.ci]: the ownership of files in the `ci` image looks completely different from the published image, so should these files be owned by root to promote test and production parity?

File: Dockerfile.composer [hunk 5]
```diff
@@ -0,0 +1,10 @@
+FROM composer:1.6.4
+
+ARG composer_dev_arg
+ARG php_version
+
+RUN composer --no-interaction config --global platform.php ${php_version}
```
Comment (giorgiosironi) [inline Dockerfile.composer:6]: looks like a `.php` file, but it isn't

**Review by giorgiosironi — APPROVED** [review]

**Comment by thewilkybarkid** [comment]
From 224 MB to 28.

**Comment by giorgiosironi** [comment]
:rocket:

**Pull request merged**